chromium/oqs-Linux.patch

```diff
@@ -0,0 +1,208 @@
++    case bssl::SignatureAlgorithm::kFalcon1024:
++    case bssl::SignatureAlgorithm::kSphincssha2256fsimple:
++    case bssl::SignatureAlgorithm::kSphincssha2256ssimple:
++    case bssl::SignatureAlgorithm::kSphincsshake256fsimple:
++    case bssl::SignatureAlgorithm::kSphincsshake256ssimple:
+       return true;
+   }
+ 
+diff --git a/net/cert/x509_certificate.cc b/net/cert/x509_certificate.cc
+index f23121ac4a..9b213e3dcc 100644
+--- a/net/cert/x509_certificate.cc
++++ b/net/cert/x509_certificate.cc
+@@ -644,6 +644,36 @@ void X509Certificate::GetPublicKeyInfo(const CRYPTO_BUFFER* cert_buffer,
+     case EVP_PKEY_DH:
+       *type = kPublicKeyTypeDH;
+       break;
++    case EVP_PKEY_DILITHIUM2:
++    case EVP_PKEY_DILITHIUM3:
++    case EVP_PKEY_DILITHIUM5:
++      *type = kPublicKeyTypeDilithium;
++      break;
++    case EVP_PKEY_FALCON512:
++    case EVP_PKEY_FALCON1024:
++      *type = kPublicKeyTypeFalcon;
++      break;
++    case EVP_PKEY_MLDSA44:
++    case EVP_PKEY_MLDSA65:
++    case EVP_PKEY_MLDSA87:
++      *type = kPublicKeyTypeMLDSA;
++      break;
++    case EVP_PKEY_SPHINCSSHA2128FSIMPLE:
++    case EVP_PKEY_SPHINCSSHA2128SSIMPLE:
++    case EVP_PKEY_SPHINCSSHA2192FSIMPLE:
++    case EVP_PKEY_SPHINCSSHA2192SSIMPLE:
++    case EVP_PKEY_SPHINCSSHA2256FSIMPLE:
++    case EVP_PKEY_SPHINCSSHA2256SSIMPLE:
++      *type = kPublicKeyTypeSPHINCSSHA2;
++      break;
++    case EVP_PKEY_SPHINCSSHAKE128FSIMPLE:
++    case EVP_PKEY_SPHINCSSHAKE128SSIMPLE:
++    case EVP_PKEY_SPHINCSSHAKE192FSIMPLE:
++    case EVP_PKEY_SPHINCSSHAKE192SSIMPLE:
++    case EVP_PKEY_SPHINCSSHAKE256FSIMPLE:
++    case EVP_PKEY_SPHINCSSHAKE256SSIMPLE:
++      *type = kPublicKeyTypeSPHINCSSHAKE;
++      break;
+   }
+   *size_bits = base::saturated_cast<size_t>(EVP_PKEY_bits(pkey.get()));
+ }
+diff --git a/net/cert/x509_certificate.h b/net/cert/x509_certificate.h
+index ad138a534e..2749f29efe 100644
+--- a/net/cert/x509_certificate.h
++++ b/net/cert/x509_certificate.h
+@@ -47,7 +47,12 @@ class NET_EXPORT X509Certificate
+     kPublicKeyTypeDSA,
+     kPublicKeyTypeECDSA,
+     kPublicKeyTypeDH,
+-    kPublicKeyTypeECDH
++    kPublicKeyTypeECDH,
++    kPublicKeyTypeDilithium,
++    kPublicKeyTypeFalcon,
++    kPublicKeyTypeMLDSA,
++    kPublicKeyTypeSPHINCSSHA2,
++    kPublicKeyTypeSPHINCSSHAKE
+   };
+ 
+   enum Format {
+diff --git a/net/quic/quic_session_pool.cc b/net/quic/quic_session_pool.cc
+index a7e1ecbe60..3ab2c9cff7 100644
+--- a/net/quic/quic_session_pool.cc
++++ b/net/quic/quic_session_pool.cc
+@@ -347,7 +347,16 @@ QuicSessionPool::QuicCryptoClientConfigOwner::QuicCryptoClientConfigOwner(
+                           base::Unretained(this)));
+   if (quic_session_pool_->ssl_config_service_->GetSSLContextConfig()
+           .PostQuantumKeyAgreementEnabled()) {
+-    config_.set_preferred_groups({SSL_GROUP_X25519_KYBER768_DRAFT00,
++    config_.set_preferred_groups({SSL_GROUP_KYBER512, SSL_GROUP_KYBER768, SSL_GROUP_KYBER1024,
```

Comment:
Worthwhile pointing in the documentation to this code as defining which groups are supported by default?